utils/bitfield/src/unvalidated.rs

```diff
@@ -17,12 +18,21 @@ impl<'a> Validate<'a> for &'a mut UnvalidatedBitField {
     fn validate(self) -> Result<&'a BitField> {
         self.validate_mut().map(|bf| &*bf)
     }
+    /// it's O(1) to get max set value in bitfield during validation, so we do that here.
```

Comment:
I would put this comment in a normal comment rather than Doc comment.
Is it really O(1)? Because if input is n bytes, it looks more like a linear complexity to me, no?